python/lsst/daf/persistence/butler.py

```diff
@@ -1164,6 +1166,29 @@ def getKeys(self, datasetType=None, level=None, tag=None):
                     break
         return keys
 
+    def getDatasetTypes(self, tag=None):
+        """Get the valid dataset types for all known repos or those matching
+        the tags.
+
+        Parameters
+        ----------
+        tag - any, or list of any
+            Any object that can be tested to be the same as the tag in a dataId passed into butler input
+            functions. Applies only to input repositories: If tag is specified by the dataId then the repo
+            will only be read from used if the tag in the dataId matches a tag used for that repository.
```

Comment:
Grammar here is funny.

Reply:
I copied it from elsewhere :).  I'll fix in both places.